DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+```java
+test().whileTrue(systemsCheck());
+```
+#### Viewing results 
+This is probably the most important part of systems check, checking if our tests passed. Using `FaultLogger` we are able to pass the results of our checks into networkTables and see them in both sim and `Elastic`. 
+- Elastic is a dashboard that's great use for drive team as it lets you see a bunch of robot data on a nice layout. You don't necessarily need this app for the project and can view this in sim. 
+
+Make sure you are periodically updating FaultLogger in `robot.java`. (Done in the robot constructor)
+```java
+addPeriodic(FaultLogger::update, 2); 
+```
+Now launch the sim window (ctrl + shift + p) -> ```simulate robot code```.
+Once it's loaded, click the `Test` mode and then open NetworkTables info.
+![](image-1.png)
+In Faults and Total Faults, we can see that we have indeed ran our checks and since there are no warnings, it is safe to assume that in this case the checks have passed. 
+- To see this in Elastic, click `Test` and then head to Elastic. Right click, click add widgets, click `Faults` and drag and drop `Total Faults` and `Active Faults` onto the layout. You should see the following if the tests have passed. 
+
+![](image-2.png)
+
+Same with Unit Tests, it's good to have system checks for all runnable parts of the robot to ensure the robot is fully ready for the game. As a notice, please make sure to run system checks while the robot is ON the robot cart. 
+### Simulation and Logging
+To finish our project up, we are going to simlute the drive we've made. Please read the [Simulation reference sheet](https://github.com/SciBorgs/SciGuides/blob/main/Simulation.md) before continuing on with this. 
+
+Setting Up the Simulation
+To simulate the drivetrain, we’re going to use the DifferentialDrivetrainSim class. This simulation will model the physical characteristics of our robot—like the motors, mass, and wheel dimensions—so we can see how the code will affect the robot in a virtual environment.
+
+Here’s how we set it up in our constructor:
+```java
+private final DifferentialDrivetrainSim driveSim;
+...
+driveSim =
+    new DifferentialDrivetrainSim(
+        DCMotor.getMiniCIM(2),
+        GEARING,
+        MOI,
+        DRIVE_MASS.in(Kilograms),
+        WHEEL_RADIUS,
+        TRACK_WIDTH,
+        STD_DEVS);
+```
+- Remember that all of these values would be stored in `driveConstants`:
+  ```
+  public static final double TRACK_WIDTH = 0.7112;
+  public static final double WHEEL_RADIUS = Units.inchesToMeters(3);
+  public static final double GEARING =  8;
+  public static final double MOI = 7.5;
+  public static final Measure<Mass> DRIVE_MASS = Kilograms.of(60.0);
+  public static final Matrix<N7, N1> STD_DEVS = VecBuilder.fill(0, 0, 0, 0, 0, 0, 0);
+  ```
+By setting up the simulation this way, we’re creating a virtual model of our robot that behaves similarly to how it would in real life.
```

Comment:
```suggestion
By setting up the simulation with these constants, we’re creating a physics model of our robot that behaves similarly to how it would in real life.
```